src/lib/adwaitarenderer.cpp

```diff
@@ -430,8 +430,8 @@ void Renderer::renderButtonFrame(const StyleOptions &options)
 
     // render
     options.painter()->drawRoundedRect(frameRect, radius, radius);
-
-    if (!options.sunken() && options.active() && options.color().isValid()) {
+    // gtk4 adwaita looks flat these days, so we wont draw the extra decor by default
+    if (!options.sunken() && options.active() && options.color().isValid() && renderBevel) {
```

Comment:
I want to have current Adwaita-qt GTK3 based and possible do a major GTK4 overhaul in the future.

Reply:
Do a new new style altogether for gtk4 and leave the gtk3 one as is ? . Anyway, libadwaita has just been release, and things are a bit different again. Perhaps it should be inspired by that.